src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -591,3 +646,288 @@ RouterExecutorEnd(QueryDesc *queryDesc)
 	queryDesc->estate = NULL;
 	queryDesc->totaltime = NULL;
 }
+
+/*
+ * Walks each TargetEntry of the query, evaluates sub-expressions without Vars.
+ */
+void ExecuteFunctions(Query *query)
+{
+	CmdType commandType = query->commandType;
+	ListCell *targetEntryCell = NULL;
+	Node *modifiedNode = NULL;
+
+	/* DELETE has no targetList */
+	if (!(commandType == CMD_INSERT) && !(commandType == CMD_UPDATE))
+	{
+		return;
+	}
+
+	foreach(targetEntryCell, query->targetList)
```

Comment:
@marcocitus Not sure why that'd be that interesting though. Let's get the basic feature in?